### ISSUE TITLE
Implement infrastructure to query GA4 account summaries

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#3103]
It should be possible to query GA4 account summaries (see #3012 for related UA account summaries issue).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `Analytics_4` module class should have a new REST datapoint `GET:account-summaries`.
    * It should rely on the [accountSummaries.list](https://developers.google.com/analytics/devguides/config/admin/v1/rest/v1alpha/accountSummaries/list) API endpoint.
    * It should return the array of account summary objects.
* The `modules/analytics-4` datastore should get a new selector `getAccountSummaries()`.
    * It should rely on the `GET:modules/analytics-4/data/account-summaries` REST datapoint to get the user's account summaries.

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by felixarntz** [comment]
@eugene-manuilov Why the `pageSize` limit on the PHP side? Is there a specific low default limit we need to override? It is only allowed to have 50 Analytics accounts per Google account, so this should never be more than 50 entries anyway.

**Comment by eugene-manuilov** [comment]
> It is only allowed to have 50 Analytics accounts per Google account, so this should never be more than 50 entries anyway.

:thinking: I read that there is a limit in 100 Analytics account per Google account and it can be increased in Analytics 360. Anyway the documentation says the following for the `pageSize` argument:

```
The maximum number of AccountSummary resources to return. The service may return fewer than this value, even if there are additional pages. If unspecified, at most 50 resources will be returned. The maximum value is 200; (higher values will be coerced to the maximum)
```

It makes me think that in theory users can have even more than 100 "summaries", so we need to be prepared :smile:

**Comment by felixarntz** [comment]
@eugene-manuilov Makes sense, let's go with that then. It would be good to add a code comment on why this is set to 200 though.

IB ✅

**Comment by johnPhillips** [comment]
@eugene-manuilov Any chance you could add a QA brief? 🙏

**Comment by eugene-manuilov** [comment]
@johnPhillips ah... yes, updated! Thanks!
